### DISCUSSION TITLE
Nested polymorphic table?

### DISCUSSION THREAD

**Discussion by rdbisme** [sqlalchemy/sqlalchemy#10810] [Usage Questions]
Hello is it possible to define nested polymorphic hierarchies with SQLAlchemy? Like a root table having a discriminator `type`, with several children tables, one of those having a subdiscriminator column `category` with nested children?

```mermaid
flowchart TD
    A[Root\nDiscriminator: type] -->B[Submodel B\n type: 'B']
    A --> C[Submodel C\n type: 'C'\n subdiscriminator: 'category']
    C --> D[Submodel D\n type: 'C'\n category: 'D']
    C --> E[Submodel E\n type: 'C'\n category: 'E']
```

**Comment by zzzeek** [comment]
we have a very old add for this feature at #2555 which has been closed as it's a complicated feature add for which there's been very little interest; it's also questionable if complex hierarchies like this should be encouraged, since we're mapping to database tables.

we can revisit the feature but we generally try to not add new complexity these days unless there's a super good reason.

**Reply by rdbisme** [reply]
Ok thanks for the quick feedback :)
